afdko/PKGBUILD

```diff
@@ -3,58 +3,63 @@
 
 pkgname=afdko
 pkgver=4.0.1
-pkgrel=1
-pkgdesc='Adobe Font Development Kit for OpenType'
+pkgrel=6
+pkgdesc="Adobe Font Development Kit for OpenType"
 arch=(x86_64)
 url="https://github.com/adobe-type-tools/$pkgname"
-license=(custom)
+license=('Apache-2.0')
 _py_deps=(booleanoperations
           brotli # for fonttools[woff]
           defcon
           fontmath
-          fontparts
           fontpens # for defcon[pens]
           fonttools
           fs # for fonttools[ufo]
-          lxml # for fonttools[lxml] and defcon[lxml]
+          lxml # for defcon[lxml] and fonttools[lxml]
           tqdm
           ufonormalizer
           ufoprocessor
           unicodedata2 # for fonttools[unicode]
           zopfli) # for fonttools[woff]
-depends=(python
-        "${_py_deps[@]/#/python-}"
-        psautohint)
+depends=(gcc-libs
+         glibc
+         libxml2
+         python
+         "${_py_deps[@]/#/python-}")
 makedepends=(cmake
              git # Upstream Issue: https://github.com/adobe-type-tools/afdko/issues/1407
-             python-setuptools-scm
-             python-scikit-build)
-checkdepends=(python-pytest)
+             ninja
+             python-{build,installer,wheel}
+             python-{scikit-build,setuptools-scm})
+checkdepends=(python-{pytest,virtualenv})
 conflicts=(spot-client)
 _archive="$pkgname-$pkgver"
-source=("$url/releases/download/$pkgver/$_archive.tar.gz")
+source=("$_archive.tar.gz::$url/releases/download/$pkgver/$_archive.tar.gz")
 sha256sums=('22dd90f0f7b4bc6eacbe8cddb0d72a484cf3f2d6cb474b0e3496de161177f019')
 
 prepare () {
-	cd "$_archive"
-	sed -i -e 's/==/>=/g;s/,<=[0-9.]\+//' requirements.txt
-	sed -i -E "/'(wheel|cmake|ninja)',?$/d" setup.py
+  cd "$_archive"
+  sed -i '/"wheel",/d;/"cmake",/d;/"ninja"/d;s/"scikit-build",/"scikit-build"/' pyproject.toml
+  sed -i -e 's/==/>=/g;s/,<=[0-9.]\+//' requirements.txt
+  sed -i -E "/'(wheel|cmake|ninja)',?$/d" setup.py
 }
 
 build() {
-	cd "$_archive"
-	python setup.py build_ext
-	python setup.py build
+  cd "$_archive"
+  python -m build --wheel --no-isolation
 }
 
 check() {
-	cd "$_archive"
-	# Upstream test suite uses vendored deps and the paths are foobared
-	# PYTHONPATH=python pytest
+  cd "$_archive"
+  virtualenv pyvenv --system-site-packages
```

Comment:
It is completely pointless to test packages in a virtualenv, this only tests the upstream test suite in isolation. The only point of a `check()` step in the build is to test *integration* with the host environment. If we can't work that out then there is no point in even running tests. Although of course it would be good to work out and run as many integrated tests as we can.

Reply:
https://github.com/adobe-type-tools/afdko/issues/1739#issuecomment-2016672006

Reply:
The `check()` stage runs after the build but without the final package being installed, so you do have to work out all the paths. System dependencies should be installed (via `makedepends`+`checkdepends`), so just the path to internal stuff needs fixing. Usually that's what adding the build directory to `PYTHON_PATH` accomplishes, but as my commented code there suggests upstream is doing vendoring and it is not trivial to untangle what they've done with tests.

Reply:
It's not just inelegant, it doesn't actually test integration with the package. It just sets up a shim between the host system, the build, and the test suite that automatically makes up any differences. Anything missing or incompatible from the host system is just going to be supplied by the virtualenv. That means it doesn't actually reveal any issues of the sort we're looking for by having a check step in the first place.

Either we work it out so we can actually run the tests on the host system such that they serve as integration tests or we do nothing it all. There is no point in faking it and just running the upstream project's unit tests on our end. We presume they are doing that upstream anyway.